test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1048,13 +1054,30 @@ private void assertSameSyncIdSameDocs() {
         }
     }
 
-    private void assertShardIndexCounter() {
+    private void assertShardIndexCounter() throws IOException {
         final Collection<NodeAndClient> nodesAndClients = nodes.values();
         for (NodeAndClient nodeAndClient : nodesAndClients) {
             IndicesService indexServices = getInstance(IndicesService.class, nodeAndClient.name);
             for (IndexService indexService : indexServices) {
                 for (IndexShard indexShard : indexService) {
-                    assertThat("index shard counter on shard " + indexShard.shardId() + " on node " + nodeAndClient.name + " not 0", indexShard.getActiveOperationsCount(), equalTo(0));
+                    int activeOperationsCount = indexShard.getActiveOperationsCount();
+                    if (activeOperationsCount > 0) {
+                        TaskManager taskManager = getInstance(TransportService.class, nodeAndClient.name).getTaskManager();
+                        DiscoveryNode localNode = getInstance(ClusterService.class, nodeAndClient.name).localNode();
+                        List<TaskInfo> taskInfos = taskManager.getTasks().values().stream()
+                            .filter(task -> task instanceof ReplicationTask)
+                            .map(task -> task.taskInfo(localNode, true))
+                            .collect(Collectors.toList());
+                        ListTasksResponse response = new ListTasksResponse(taskInfos, Collections.emptyList(), Collections.emptyList());
+                        XContentBuilder builder = XContentFactory.jsonBuilder()
```

Comment:
not sure if this should be pretty printed as we put in the exception message.  we'll have to see an example to be sure. Alternatively we can log a warning line like we do in ensure green. Also, you can consider using Strings.toString().